src/matrix/room/Room.js

```diff
@@ -110,6 +110,18 @@ export class Room extends BaseRoom {
         }
     }
 
+    async fetchContext(eventId, log = null) {
```

Comment:
This method should probably:
 - return the event entry for given event id? this way it can be used for reply preview rendering, I think?
 - first check if the event is not already in storage

Reply:
Since we have decided out of band to make `fetchContext` private and call it from `observeEvent`, we can probably forego checking if the entry is already in storage (`observeEvent` would already make that check)